test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,328 @@
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/main/test.h"
+#include "src/main/validator.h"
+
+namespace test_harness {
+
+/* This test produces a workload that encourages the background compaction server to do work by:
+ *      1. Performing random truncations that remove 20% of keys on randomly selected tables.
+ *      2. Providing a "maintenance window" which allows compact to continue running while all other
+ *         operations are paused. The period of the maintenance window is set by the custom
+ *         operations op_rate.
+ *      3. Performing inserts to ensure the files continue to grow.
+ */
+class background_compact : public test {
+    bool maintenance_window = false;
+
+public:
+    background_compact(const test_args &args) : test(args)
+    {
+        init_operation_tracker();
+    }
+
+    /* Custom operation to simulate toggling maintenance windows in a workload. */
+    void
+    custom_operation(thread_worker *tw) override final
+    {
+        std::string log_prefix =
+          type_string(tw->type) + " thread {" + std::to_string(tw->id) + "}: ";
+        logger::log_msg(
+          LOG_INFO, type_string(tw->type) + " thread {" + std::to_string(tw->id) + "} commencing.");
+
+        while (tw->running()) {
+            maintenance_window = !maintenance_window;
+            std::string state = maintenance_window ? "On" : "Off";
+            logger::log_msg(LOG_TRACE, log_prefix + " toggle maintenance window " + state);
+
+            tw->sleep();
+        }
+    }
+
+    void
+    remove_operation(thread_worker *tw) override final
+    {
+        std::string log_prefix =
+          type_string(tw->type) + " thread {" + std::to_string(tw->id) + "}: ";
+        logger::log_msg(LOG_INFO, log_prefix + "commencing.");
+
+        std::map<uint64_t, scoped_cursor> rnd_cursors, stat_cursors;
+
+        /* Loop while the test is running. */
+        while (tw->running()) {
+            /* Make sure we're not doing any work during the maintenance window. */
+            if (maintenance_window) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
You can use `std::chrono::seconds` here instead.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9691/commits/b21630dc0d16a3e94d7d729cc9d519ab60c84c7e